### ISSUE TITLE
Make `offline-plugin` work with webpack-dev-server

### ISSUE THREAD

**Issue by NekR** [NekR/offline-plugin#138]
This includes live reload and Hot Module Replacement. In those cases `offline-plugin`'s cache functions should do nothing, i.e. not handle `fetch` event.

Reasons: someone may want to test push notifications / background sync in dev mode with webpack-dev-server.

(Probably even reload SW file's code when it changes)

**Comment by SpaceK33z** [comment]
Let me know if you need any hooks from webpack-dev-server (I'm the maintainer).

**Comment by NekR** [comment]
@SpaceK33z Yeah, I know :-) Will do. Thanks for offering help, that's really appreciated 👍 

I have a friend willing to take over this issue, so I created it for him. Not sure when this will happen though. I'll let you know if something will be needed (though I hope everything is there already).

**Comment by NekR** [comment]
The difference between normal compilation and `webpack-dev-server` compilation: https://www.diffchecker.com/MSt07Pj1
Seems pretty trivial to check for `MemoryFilesystem` usage and react appropriately.

/cc @SpaceK33z I know it isn't bullet-proof detection way but seems pretty solid to me, given that I don't know any other webpack tools using memory fs. What you think?

**Comment by SpaceK33z** [comment]
@NekR yeah that seems like a good idea for now. Note that `MemoryFilesystem` is set in webpack-dev-middleware, so it will also detect users that only use the middleware.

**Comment by NekR** [comment]
> Note that MemoryFilesystem is set in webpack-dev-middleware, so it will also detect users that only use the middleware.

@SpaceK33z 
Yeah, I think that's what I want + a flag to turn normal behavior of the plugin back even when `MemoryFilesystem` is used. Thanks for info 👍
